### PR TITLE
feat: wallet integration pop call chain

### PR DESCRIPTION
**Pull request by AlexD10S**
Pointing to a temporal branch until https://github.com/r0gue-io/pop-cli/pull/365 gets rebase from latest changes in [peter/feat-wallet-integration](https://github.com/r0gue-io/pop-cli/tree/peter/feat-wallet-integration)

[sc-1721]

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -57,7 +57,6 @@ contract-extrinsics = { git = "https://github.com/use-ink/cargo-contract", branc
 #contract-transcode = "5.0.0-alpha"
 contract-transcode = { git = "https://github.com/use-ink/cargo-contract", branch = "peter/chore-make-types-pub" }
 heck = "0.5.0"
-hex = { version = "0.4.3", default-features = false }
```
Comment (AlexD10S) [inline Cargo.toml:60]: Removed for consistency. In the main feat PR other libraries.

File: crates/pop-cli/src/commands/up/contract.rs [hunk 2]
```diff
@@ -338,29 +337,6 @@ impl UpContractCommand {
 		Ok(())
 	}
 
-	/// Handles the optional termination of a local running node.
-	fn terminate_node(process: Option<(Child, NamedTempFile)>) -> anyhow::Result<()> {
```
Comment (AlexD10S) [inline crates/pop-cli/src/commands/up/contract.rs:342]: Refactor: moved into  `common/contracts`

File: crates/pop-cli/src/commands/up/contract.rs [hunk 3]
```diff
@@ -381,33 +357,6 @@ impl UpContractCommand {
 			Ok((call_data, hash))
 		}
 	}
-
-	async fn wait_for_signature(&self, call_data: Vec<u8>) -> anyhow::Result<Option<String>> {
```
Comment (AlexD10S) [inline crates/pop-cli/src/commands/up/contract.rs:385]: Refactor: Will be used in more than one place. Moved  into a new file `common/wallet`

File: crates/pop-cli/src/commands/up/contract.rs [hunk 4]
```diff
@@ -494,82 +440,6 @@ mod tests {
 		Ok(())
 	}
 
-	#[test]
-	fn has_contract_been_built_works() -> anyhow::Result<()> {
```
Comment (AlexD10S) [inline crates/pop-cli/src/commands/up/contract.rs:498]: No needed anymore

File: crates/pop-cli/src/commands/up/contract.rs [hunk 5]
```diff
@@ -494,82 +440,6 @@ mod tests {
...
-		assert!(has_contract_been_built(Some(&path.join(name))));
-		Ok(())
-	}
-
-	// TODO: delete this test.
-	// This is a helper test for an actual running pop CLI.
-	// It can serve as the "frontend" to query the payload, sign it
-	// and submit back to the CLI.
-	#[tokio::test]
-	async fn sign_call_data() -> anyhow::Result<()> {
```
Comment (AlexD10S) [inline crates/pop-cli/src/commands/up/contract.rs:523]: Refactor: Moved into `common/wallet`

File: crates/pop-cli/src/commands/call/chain.rs [hunk 6]
```diff
@@ -411,14 +468,45 @@ impl Call {
...
+		}
 		if self.sudo {
 			full_message.push_str(" --sudo");
 		}
 		full_message
 	}
 }
 
+// Sign and submit an extrinsic using wallet integration.
+async fn submit_extrinsic_with_secure_signing(
```
Comment (peterwht) [inline crates/pop-cli/src/commands/call/chain.rs:485]: maybe "_with_wallet" is more consistent?

File: crates/pop-cli/src/commands/call/chain.rs [hunk 7]
```diff
@@ -40,6 +43,9 @@ pub struct CallChainCommand {
 	/// - with a password "//Alice///SECRET_PASSWORD"
 	#[arg(short, long)]
 	suri: Option<String>,
+	/// Use your browser wallet to sign the extrinsic.
+	#[arg(name = "use-wallet", long, default_value = "false", conflicts_with = "suri")]
```
Comment (peterwht) [inline crates/pop-cli/src/commands/call/chain.rs:47]: use `short('w')`

**Review by peterwht — COMMENTED** [review]
Did a quick skim. Looking pretty good so far. Seems like one of the more simple PRs thanks to the setup of call parachain.

Will do a more thorough review tomorrow

File: crates/pop-cli/src/commands/call/chain.rs (hunk 6 above)
Comment (AlexD10S) [reply crates/pop-cli/src/commands/call/chain.rs]: You are right, makes more sense use wallet here.

File: crates/pop-cli/src/commands/call/chain.rs (hunk 7 above)
Comment (AlexD10S) [reply crates/pop-cli/src/commands/call/chain.rs]: https://github.com/r0gue-io/pop-cli/pull/379/commits/7b8692497b140e64e35298cd25a9775c001a15fb

**Review by peterwht — APPROVED** [review]

**Pull request merged**